From 0fe583a0a3eda9cab9f9f4c96cfa27435882473d Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Tue, 19 Jul 2022 14:49:53 +0800
Subject: [PATCH 34/62] Revert "drm/msm/dp: fix event thread stuck in
 wait_event after kthread_stop()"

This reverts commit a10092dabae6a3536c60c9c0b16d1135aba0ef74.
---
 drivers/gpu/drm/msm/dp/dp_display.c | 13 ++++---------
 1 file changed, 4 insertions(+), 9 deletions(-)

diff --git a/drivers/gpu/drm/msm/dp/dp_display.c b/drivers/gpu/drm/msm/dp/dp_display.c
index ebd05678a27b..09c8e50da68d 100644
--- a/drivers/gpu/drm/msm/dp/dp_display.c
+++ b/drivers/gpu/drm/msm/dp/dp_display.c
@@ -998,20 +998,15 @@ static int hpd_event_thread(void *data)
 
 	dp_priv = (struct dp_display_private *)data;
 
-	while (1) {
+	while (!kthread_should_stop()) {
 		if (timeout_mode) {
 			wait_event_timeout(dp_priv->event_q,
-				(dp_priv->event_pndx == dp_priv->event_gndx) ||
-					kthread_should_stop(), EVENT_TIMEOUT);
+				(dp_priv->event_pndx == dp_priv->event_gndx),
+						EVENT_TIMEOUT);
 		} else {
 			wait_event_interruptible(dp_priv->event_q,
-				(dp_priv->event_pndx != dp_priv->event_gndx) ||
-					kthread_should_stop());
+				(dp_priv->event_pndx != dp_priv->event_gndx));
 		}
-
-		if (kthread_should_stop())
-			break;
-
 		spin_lock_irqsave(&dp_priv->event_lock, flag);
 		todo = &dp_priv->event_list[dp_priv->event_gndx];
 		if (todo->delay) {
-- 
2.27.0

